src/pytorch_lightning/core/saving.py

```diff
@@ -156,6 +158,11 @@ def _load_from_checkpoint(
     with pl_legacy_patch():
         checkpoint = pl_load(checkpoint_path, map_location=map_location)
 
+    # convert legacy checkpoints to the new format
+    checkpoint = _pl_migrate_checkpoint(
+        checkpoint, checkpoint_path=(checkpoint_path if isinstance(checkpoint_path, (str, Path)) else None)
```

Comment:
The isisntance is to pass None when `checkpoint_path` is an IO type? Shouldnt't that also be supported?

Reply:
Can't we say the same? We don't have IO specific logic, the annotation is there because it's in `torch.load`

Reply:
But if checkpoint_path is bytes, we will dump a bunch of bytes into the info message. What is the usefulness of that?

Reply:
I see. Resolving